app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -468,49 +467,82 @@ class ChatRoomPresenter @Inject constructor(
 
     fun uploadImage(roomId: String, mimeType: String, uri: Uri, bitmap: Bitmap, msg: String) {
         launchUI(strategy) {
-            view.showLoading()
+            val fileName = uriInteractor.getFileName(uri) ?: uri.toString()
             try {
                 withContext(Dispatchers.Default) {
-                    val fileName = uriInteractor.getFileName(uri) ?: uri.toString()
                     if (fileName.isEmpty()) {
                         view.showInvalidFileMessage()
                     } else {
                         val byteArray =
                             bitmap.getByteArray(mimeType, 60, settings.uploadMaxFileSize())
-                        retryIO("uploadFile($roomId, $fileName, $mimeType") {
-                            client.uploadFile(
-                                roomId,
-                                fileName,
-                                mimeType,
-                                msg,
-                                description = fileName
-                            ) {
-                                byteArray.inputStream()
-                            }
-                        }
 
+                        val id = UUID.randomUUID().toString()
+                        val username = userHelper.username()
+                        val user = userHelper.user()
+
+                        val attachment = Attachment(title=fileName, type="file", description = fileName, titleLink=uri.toString(), titleLinkDownload = true, imageUrl = uri.toString())
```

Comment:
This line reverts the change we made in this PR https://github.com/WideChat/Rocket.Chat.Kotlin.SDK/pull/19 by showing the fileName and the description.

Reply:
That changes are honoured here https://github.com/WideChat/Rocket.Chat.Kotlin.SDK/pull/20/files#diff-26b8c32e93ce0cfbb97ae5cdc53f48b3R257

Reply:
But not in the case when the initial upload fails and we use this "temporary" msg object to populate the room.

Reply:
Got it, will fix this.